### ISSUE TITLE
Get details of ConstraintViolationError

### ISSUE THREAD

**Issue by divan** [edgedb/edgedb-go#269]
Is it possible to extract information about constraint violation error from `edgedb.Error`?

Idea is that I'm interested in knowing which field exactly has violated constraints (say, is it an `email` field of User or something else), so the different actions can be taken.

Seems like right now the only way is to parse a string "email violates exclusivity constraint", which is not reliable way in a long term.

**Comment by fmoor** [comment]
I don't think that there is structured information like that by default. I may be wrong though. I'll double check tomorrow. However, you can override the error message with whatever you want including structured data like json. You will still have to parse it, but its better than parsing an error message.

**Comment by fmoor** [comment]
Here is a link to the [relevant documentation](https://www.edgedb.com/docs/datamodel/inheritance#constraints). `errmessage` is the bit the changes the message.

**Comment by fmoor** [comment]
I confirmed that the server doesn't send any structured data that indicates which constraint has been violated.

**Comment by divan** [comment]
@fmoor thanks! Not high a priority, but would be nice to have it someday to make error handling more reliable.

**Comment by fmoor** [comment]
For whoever stumbles on this issue, we plan to refactor errors in EdgeQL it self. This will improve a lot of things. Information about constraint violations being one of them.

**Comment by fmoor** [comment]
I'm going to close this because there is no way to improve the situation until errors are refactored in the query language.
